### DISCUSSION TITLE
how to trace with spring kafka without agent ?

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-java-instrumentation#6332] [Q&A] [answered]
Hello,

I would want to trace kafka messages with spring kafka without the agent, how can I do this ?

How can I set context propagation in headers with spring kafka ? 

Thank you for your time.

Regards,

Dimitri

**Comment by trask** [comment, marked as answer]
hi @dimitri995! this was just implemented recently and will be in the upcoming release this week: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6283

**Reply by dimitri995** [reply]
Thank you for your response @trask. Is it possible to provide an example about the use of the spring kafka library tracing with the new release ? 
Thank you for your time !

**Reply by trask** [reply]
check out the test for now: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-kafka-2.7/library/src/test/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/SpringKafkaNoReceiveTelemetryTest.java

**Reply by mateuszrzeszutek** [reply]
Hey @dimitri995 ,
The spring-kafka instrumentation only instruments the consumer side, you'll have to include the kafka-clients-2.6 instrumentation too, for the producer side. The link Trask posted above contains an example that sets both of them up.

**Reply by dimitri995** [reply]
Hey @mateuszrzeszutek , @trask 

Thank you for your response. It's working fine now. But, do you know if there is an integration with the Schema Registry  ?  Like if you produce a message in Avro format but this produced message is not conform to the target schema. The trace is send but the message is not. Then in my backend I can see the trace like it was sent..
OpenTelemetry can handle this ? 

Thank your for your time !

**Reply by mateuszrzeszutek** [reply]
Currently there's not support for schema registries in the Kafka instrumentation. I don't really know how that works internally for Kafka, but perhaps it should just fail the producer span and set the status to error? Anyway, contributions are welcome 🙏
